### ISSUE TITLE
VM Auth Regex for URL prefix endpoints (Select/Insert) 

### ISSUE THREAD

**Issue by sham-github** [VictoriaMetrics/VictoriaMetrics#1064]
**Is your feature request related to a problem? Please describe.**
we want to use same credentials for a user in VM-auth for select and insert URL , so we are looking for a workaround where we can pass it like below.
- username: "abc_user"
  password: "abc_pass1"
  url_prefix: http://vic-met-victoria-metrics-cluster-vminsert:8481/insert/1/prometheus

- username: "abc_user"
  password: "abc_pass2"
  url_prefix: http://vic-met-victoria-metrics-cluster-vminsert:8481/insert/1/prometheus

**Describe the solution you'd like** below should work 
- username: "abc_user"
  password: "abc_pass"
  url_prefix: "http:\/\/vic-met-victoria-metrics-cluster-vm(select|insert):(8480|8481)\/(select|insert)\/1/prometheus"

**Comment by valyala** [comment]
I think it would be better introducing some type of URL mapping. For example, the following config would route requests for  `/api/v1/query` and `/api/v1/query_range` paths to `vmselect`, while requests for `/api/v1/write` path will be routed to `vminsert`:

```yml
- username: foo
  password: bar
  url_map:
  - src_paths: ["/api/v1/query", "/api/v1/query_range"]
    url_prefix: "http://vmselect/select/1/prometheus"
  - src_paths: ["/api/v1/write"]
    url_prefix: "http://vminsert/insert/1/prometheus"
```

**Comment by valyala** [comment]
The url mapping described in the message above has been implemented in the commit 1e38ad6d20c6afb0f35dde472cbbf226ed03bcd5 . This allows routing requests to `vmauth` from a single user to multiple targets depending on the request path. @sham-github , try playing with this functionality by using `vmauth` built from the commit mentioned above. See [build instructions for vmauth](https://victoriametrics.github.io/vmauth.html#how-to-build-from-sources).

**Comment by valyala** [comment]
The `url_map` feature described at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1064#issuecomment-777069051 is available starting from [v1.54.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the feature request as done.
